src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs

```diff
@@ -234,6 +244,14 @@ public string FilePath
         {
             get
             {
+                if (_hostType == HostType.TypeScript)
+                {
+                    // Append a .js extension to the path 
+                    // so that the contained document will  
+                    // be processed as JavaScript.
+                    return Key.Moniker + JSFileExtension;
```

Comment:
It appears that you're deciding the host type based on content type, but that doesn't need to be controlled here then. Your actual classifiers and other editor entrypoints can look at the content type there and pass things through in whatever way you want. You shouldn't need to change this.